crates/dojo-lang/src/introspect/layout.rs

```diff
@@ -27,7 +27,7 @@ pub fn build_field_layouts(
             }
 
             let field_name = m.name(db).text(db);
-            let field_selector = get_selector_from_name(field_name.as_str()).unwrap().to_string();
+            let field_selector = compute_bytearray_hash(&field_name.to_string()).to_string();
```

Comment:
I'm wondering if for the fields we should switch to `ByteArray`. Starknet is still using the selector (snkeccak, `get_selector_by_name` equivalent: https://github.com/starkware-libs/cairo/blob/72d1d2f50e8c2c13d5414cc4365683299d8f3c8b/crates/cairo-lang-starknet/src/plugin/storage.rs#L270).

The only reason I can see to use the byte array is to ensure we can support any size of field name. But 31 chars is already super long for a field name I guess.

wdyt @remybar?

Reply:
Ah yes, you're right !
My main concern was that it's not possible to compute this selector in Cairo with a string stored in a variable (so, not hardcoded), but  in our Cairo contracts we never compute this selector from field name, we just use the computed value.